### ISSUE TITLE
Tinkers Compat

### ISSUE THREAD

**Issue by NillerMedDild** [NillerMedDild/Enigmatica6#2331]
Growables:
- Slimy Ferns
- Saplings
- Bloodshroom
- Slimy Vine

Substrates
- Slimy Grass types

Trees:
- Greenheart
- Skyroot
- Bloodshroom

Items to hide:
- All but 1 Tinker Station
- All but 1 Part Builder
- All but 1 Crafting Station

Akashic Tome: 
- All 5 Tinkers books

**Comment by pm065** [comment]
Can you assign me, please? I may as well do this stuff.

**Comment by pm065** [comment]
- [x] Add metal fluids
- [x] Add quests
- [x] Change piggybackpack recipe to make sense
- [ ] Finish seecret (may take some time, please bear with me)
- [ ] Add growables and substrates to the crops const
- [ ] Add trees to the wood registry
- [ ] Hide unnecessary stations
- [ ] Add fluid recipe generation script
- [ ] Add expert mode extras relating to Tinker's
- [ ] Add gem fluids
- [ ] Add Molten Blizz/Basalz/Blitz
- [ ] Switch bonus Froststeel recipe to use Molten Blizz, Molten Cobalt and Molten Iron in the Seared Smeltery or Scorched Alloyer
- [ ] Add more efficient alloy recipes for Smeltery
- [ ] Add anything else I think of

**Comment by pm065** [comment]
Urgh, this PR is taking ages, so it's going to have loads of conflicts, since I'm changing things left, right and centre...

**Comment by NillerMedDild** [comment]
Do you think you'll be able to finish it before 13th of June? :) If not I'd like to chip in.

**Comment by Darkere** [comment]
Make it multiple PR's. Much better for everyone involved.

**Comment by pm065** [comment]
> Do you think you'll be able to finish it before 13th of June? :) If not I'd like to chip in.

Yes.

By the way, it is multiple, but the first is the biggest and is nearly finished :+1:

**Comment by pm065** [comment]
The main problem, honestly, is that my laptop is a tater and takes ten centuries to boot up E6 and another fifteen to open a world :/

**Comment by pm065** [comment]
Okay, new plan for the order of the PRs:
- [ ] Foundation
- [ ] Unification
- [ ] Addendum (moved to higher priority)
- [ ] Gemstones (not as important as Addendum)
- [ ] Expertise (Expert Mode is nowhere near done so it's fine to defer this till last)

**Comment by pm065** [comment]
Update: Foundation is so close to completion! I just need to find a blood island for something. I think I started the fork before BYG nether stuff was disabled, which interfered with the island generation, so I had to restart the game to change the config (:

**Comment by NillerMedDild** [comment]
Btw
>  Add more efficient alloy recipes for Smeltery

I don't think this is something we want tbh. If anything Smeltery recipes should be less efficient.

**Comment by pm065** [comment]
> Btw
> 
> > Add more efficient alloy recipes for Smeltery
> 
> I don't think this is something we want tbh. If anything Smeltery recipes should be less efficient.
> 
> My priorities are the things mentioned in the top post, tbh :) everything else can wait for after 0.5.0 if needed.

Okay! I will take this into consideration. I feel as if custom materials are important, though. The things at the top will be added VERY soon. Also, I've pretty much finished Foundation now. Expect the PR soon(tm)! I just need to add the compat things at the top, which will take pretty much zero time compared to everything else haha

**Comment by DarkAngel22579** [comment]
>Change piggybackpack recipe to make sense

Does this really need to be changed? I feel like this is changing it just for the sake of changing something

**Comment by pm065** [comment]
> > Change piggybackpack recipe to make sense
> 
> Does this really need to be changed? I feel like this is changing it just for the sake of changing something

I don't like pig iron ingot + saddle recipe, but that's just my personal preference. I'll comment it out for now.

**Comment by pm065** [comment]
Alright, so the PR is finished, pretty much. Only problem:
- _It will explode_. I've made changes to the Adventure quest chapter, for example... and so has Mute, and since I don't know how to change which branch is the upstream origin (by default it's master, it seems), I can't obtain that change. I have no idea how to handle that issue, other than recreating Mute's quests on my branch and effectively overwriting them. :/

**Comment by theboo** [comment]
Dev, always PR to and pull from dev branch... Niller is the only one who touches master and usually just right before a release

**Comment by TemperedFool** [comment]
@pm065 you may need to stash all your changes, swap branches, fetch all the diffs, and restore your stashed changes. Will be a pita either way.

**Comment by pm065** [comment]
> @pm065 you may need to stash all your changes, swap branches, fetch all the diffs, and restore your stashed changes. Will be a pita either way.

Okay, I'll see what I can do. `how do i stash my changes?`

**Comment by TemperedFool** [comment]
With the Github Desktop app (and probably any other gui app) it's a menu command - any files that have been changes will get "stashed" so you can sync things. If you're using the command line you need someone with a bigger brain than mine.

**Comment by pm065** [comment]
> With the Github Desktop app (and probably any other gui app) it's a menu command - any files that have been changes will get "stashed" so you can sync things. If you're using the command line you need someone with a bigger brain than mine.

Yep, I'm using desktop. I haven't got enough brain cells to use command line either. I'll try that, but thanks to my cursed -2 brain's stupid mistake it'll be tomorrow when I submit the PR...

**Comment by pm065** [comment]
No, wait, the problem is actually slightly simpler, I think. Even though I created the branch from develop, it still thinks master is the upstream branch it should pull changes from. How do I switch it?

**Comment by pm065** [comment]
No, wait, the problem is actually slightly simpler, I think. Even though I created the branch from develop, it still thinks master is the upstream branch it should pull changes from.
I could have grabbed the changes on my own develop branch and then merged it into the Tinker's branch, but for some reason the minecraftinstance.json file is preventing me from doing so, and it's not letting me overwrite it with the one in develop. `my life is excellent`

**Comment by MuteTiefling** [comment]
You'll need to close curseforge to be able to clear that minecraft instance file

**Comment by pm065** [comment]
> You'll need to close curseforge to be able to clear that minecraft instance file

Ah, so that's it. Thanks!
This is why nobody should ever trust me to do anything :/

**Comment by pm065** [comment]
Alright, so the PR's pretty much done. I just need to iron out a few things.
HOWEVER, I did something slightly stupid and ill-advised. In retrospect, I should have waited till the end to do this.
I wrote a Patchouli guide to Tinker's Construct, which is far, FAR more in-depth and contains every feature I know of (unlike the extremely WIP Mantle books.)
Disadvantages: Potentially controversial, and does not have the auto-generated material info pages (they would have to be added manually.)

I can see three ways to use this book:
1. Throw it out, pretend it never existed (I will adapt it to the needs of the modpack I'm making myself)
2. Keep it AND the Mantle books. Pretend to advertise the Patchouli book as a more in-depth beginner's guide that doesn't contain material info. Direct players to the Mantle books for material information.
3. Throw out the Mantle books, and add a manual material info section to the Patchouli book.

Which would be the best? Note that I will **not**, in any way, be offended by any decisions to take the first option. I respect that this pack, unlike my own (I literally renamed Botania's Mana so as to prevent confusion with Ars Nouveau's 🙃 which turned out better than I'd thought it would) does not allow extremely massive, sweeping changes that rock the foundations of each mod. If this is counted as one of those, please feel free to yeet away!

**Comment by DarkAngel22579** [comment]
> I wrote a Patchouli guide to Tinker's Construct, which is far, FAR more in-depth and contains every feature I know of (unlike the extremely WIP Mantle books.)

Definitely jumped the gun on this, as the Tinkers books are almost finished, and will almost certainly be fully fleshed out by the time 0.5.0 drops.

**Comment by ShinyPorygon** [comment]
So, as the tinkers book writer this is highly insulting. I've been waiting on a feature to finish the books that I've been writing. And I have almost finished copies on my desktop. Just have to finish a few more things get a bug worked out. 

I'd also like to point out that I looked through your  books and most of your info is **HIGHLY** incorrect. It seems as you have not even touched the 1.16 version, kept up with the updates, or anything. 

You also either haven't thought of it or didn't much care that as a result of being in such a popular pack this would get reflected and blamed on us (and then me as the book writer) Your best thing is to scrap this, because you'd have to redo 90% of it for it to be correct.

**Comment by pm065** [comment]
I am EXTREMELY sorry for insulting you. That was NOT my intention. Why did I do this? I'm an absolute idiot. f course this would happen!

My **sincerest** apology. I will remove it immediately.

**Comment by ShinyPorygon** [comment]
With all due respect it wasn't that bad. But you also were seeming to use an older version of tinkers which is where most of the mistakes come from. I personally can't even keep up with changes sometimes. Don't feel too bad. I have bad memory so I won't remember this in 5 days 😛 If anything I'm probably the nicer of people who would have said something. No hard feelings I hope. You should just try and remember to use the latest versions (as looking at your versions you used the books were way less incomplete yes. They are better now but not finished.)

**Comment by MuteTiefling** [comment]
Updates crops, soils, trees, and wood variants for tinkers.
Adds back missing slime block tag that broke our bees.
Add tooltips to tinker workstations. Still working on hiding them. Normal hide script doesn't work.

Akashic tome will take a change on Tinker's or Akashic's side.

**Comment by MuteTiefling** [comment]
ok, only thing left from the main list is the Akashic tome

**Comment by natebc** [comment]
Good work you guys!!!
